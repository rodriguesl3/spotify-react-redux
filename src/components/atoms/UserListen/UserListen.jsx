import React from 'react';

import { FaRegPlayCircle, FaRegPauseCircle } from 'react-icons/fa';
import { MdSkipPrevious, MdSkipNext } from 'react-icons/md';

import { RatingStart } from '../RatingStar/RatingStar';
import LoaderHOC from '../Loading/LoaderHOC';

const UserListen = ({ userListening, onChangeMusic }) => {
  const changeMusicHandle = (changeParam) => {
    onChangeMusic(changeParam);
  };

  return (
    <div>
      <span>
        Device:
        {userListening.device.name}
      </span>
      <p>
        Type:
        {userListening.device.type}
      </p>
      <div className="row">
        <div className="col-9">
          <span>{userListening.item && userListening.item.name}</span>
        </div>
      </div>
      <div className="row">
        <div className="col-3">
          <p className='previous' onClick={changeMusicHandle.bind(this, 'previous')}>
            <MdSkipPrevious />
          </p>
        </div>
        <div className="col-3">
          <p>
            {!userListening.is_playing ? <FaRegPlayCircle /> : <FaRegPauseCircle />}
          </p>
        </div>
        <div className="col-3">
          <p className='next' onClick={changeMusicHandle.bind(this, 'next')}>
            <MdSkipNext />
          </p>
        </div>
      </div>
      <span><RatingStart ratingValue={userListening.item && userListening.item.popularity} /></span>
    </div>
  )
};


export default UserListen;
