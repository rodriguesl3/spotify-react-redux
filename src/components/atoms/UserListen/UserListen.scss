.card-player {
  color: $base-color;
  border-radius: 25px;
  width: 100%;
  background: white;
  display: flex;
  align-items: center;
  justify-content: center;
}