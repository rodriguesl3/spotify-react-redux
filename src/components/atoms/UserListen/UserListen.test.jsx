import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import UserListen from './UserListen';

test('clicking next music.', () => {
  const onMockChangeMusic = jest.fn();
  const userListenMock = {
    item: { name: 'teste' },
    device: {
      name: 'huawei',
      type: 'mobile',
    },
  };

  const output = shallow(<UserListen
    userListening={userListenMock}
    onChangeMusic={onMockChangeMusic}
  />);

  expect(shallowToJson(output)).toMatchSnapshot();

  output.find('.next').simulate('click');
  expect(onMockChangeMusic).toBeCalledWith('next');
});


test('clicking previous music.', () => {
  const onMockChangeMusic = jest.fn();
  const userListenMock = {
    item: { name: 'teste' },
    device: {
      name: 'huawei',
      type: 'mobile',
    },
  };

  const output = shallow(<UserListen
    userListening={userListenMock}
    onChangeMusic={onMockChangeMusic}
  />);

  expect(shallowToJson(output)).toMatchSnapshot();

  output.find('.previous').simulate('click');
  expect(onMockChangeMusic).toBeCalledWith('previous');
});