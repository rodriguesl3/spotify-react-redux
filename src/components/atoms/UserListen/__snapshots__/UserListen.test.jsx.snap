// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicking next music. 1`] = `
<div>
  <span>
    Device:
    huawei
  </span>
  <p>
    Type:
    mobile
  </p>
  <div
    className="row"
  >
    <div
      className="col-9"
    >
      <span>
        teste
      </span>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <p
        className="previous"
        onClick={[Function]}
      >
        <MdSkipPrevious />
      </p>
    </div>
    <div
      className="col-3"
    >
      <p>
        <FaRegPlayCircle />
      </p>
    </div>
    <div
      className="col-3"
    >
      <p
        className="next"
        onClick={[Function]}
      >
        <MdSkipNext />
      </p>
    </div>
  </div>
  <span>
    <RatingStart />
  </span>
</div>
`;

exports[`clicking previous music. 1`] = `
<div>
  <span>
    Device:
    huawei
  </span>
  <p>
    Type:
    mobile
  </p>
  <div
    className="row"
  >
    <div
      className="col-9"
    >
      <span>
        teste
      </span>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <p
        className="previous"
        onClick={[Function]}
      >
        <MdSkipPrevious />
      </p>
    </div>
    <div
      className="col-3"
    >
      <p>
        <FaRegPlayCircle />
      </p>
    </div>
    <div
      className="col-3"
    >
      <p
        className="next"
        onClick={[Function]}
      >
        <MdSkipNext />
      </p>
    </div>
  </div>
  <span>
    <RatingStart />
  </span>
</div>
`;
